webui/react/src/pages/TaskLogs.tsx

```diff
@@ -1,108 +1,144 @@
 import queryString from 'query-string';
-import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { useParams } from 'react-router-dom';
 
-import LogViewer, { LogViewerHandles, TAIL_SIZE } from 'components/LogViewer';
+import LogViewer, { FetchConfig, FetchType } from 'components/LogViewer/LogViewer';
+import Page from 'components/Page';
 import { commandTypeToLabel } from 'constants/states';
-import usePolling from 'hooks/usePolling';
-import useRestApi from 'hooks/useRestApi';
+import useSettings from 'hooks/useSettings';
 import { paths } from 'routes/utils';
-import { getTaskLogs } from 'services/api';
-import { TaskLogsParams } from 'services/types';
-import { CommandType, Log } from 'types';
+import { detApi } from 'services/apiConfig';
+import { jsonToTaskLog } from 'services/decoder';
+import { consumeStream } from 'services/utils';
+import { CommandType } from 'types';
+
+import TaskLogFilters, { Filters } from './TaskLogFilters';
+import css from './TaskLogs.module.scss';
+import settingsConfig, { Settings } from './TaskLogs.settings';
 
 interface Params {
   taskId: string;
   taskType: string;
 }
 
-interface Queries {
-  id?: string;
-}
+type OrderBy = 'ORDER_BY_UNSPECIFIED' | 'ORDER_BY_ASC' | 'ORDER_BY_DESC';
```

Comment:
yeah this is a good call and been meaning to do this. Something like:
```
export enum OrderByFromApi {
  Asc = 'ORDER_BY_ASC',
  Desc = 'ORDER_BY_DESC',
  Unspecified = 'ORDER_BY_UNSPECIFIED',
}
```
Will create a separate ticket and PR to do this though since it touches a bunch of files.